docs/icicle/primitives/poseidon.md

```diff
@@ -0,0 +1,233 @@
+# Poseidon
+
+Poseidon hash is a popular hash in the ZK ecosystem primarily because its optimized to work over large prime fields, a common setting for ZK proofs, thereby minimizing the number of multiplicative operations required.
+
+Poseidon hash function has also been specifically designed to be efficient when implemented within ZK circuits, Poseidon hash uses far less constraints compared to other hash functions like Keccak or SHA-256 in the context of ZK circuits.
+
+Poseidon hash has been used in many popular ZK protocols such as Filecoin and [Plonk](https://drive.google.com/file/d/1bZZvKMQHaZGA4L9eZhupQLyGINkkFG_b/view?usp=drive_open).
+
+Our implementation of [Poseidon hash](https://eprint.iacr.org/2019/458.pdf) is implemented  in accordance with the optimized [Filecoin version](https://spec.filecoin.io/algorithms/crypto/poseidon/).
```

Comment:
I would move the link to the Poseidon paper to an earlier (if not the first) mention of Poseidon hash. Its especially odd here where we are saying "Our implementation" a the beginning of the sentence and not linking to our implementation

Reply:
agree, thanks